### ISSUE TITLE
Export Feature

### ISSUE THREAD

**Issue by luke-** [humhub/wiki#260]
TBD

**Comment by marc-farre** [comment]
@luke- @samuk What file format have you in mind?
- Markdown: easy for the text, but needs to be backed in a zip file if we want to export the files too (such as images)
- ODT: requires a library, e.g. https://github.com/PHPOffice/PHPWord (we already include `phpoffice/phpspreadsheet` in HumHub)

We already can export in PDF:

![image](https://github.com/user-attachments/assets/c1ca27d1-79df-42dd-8e54-52b766fa026a)

![image](https://github.com/user-attachments/assets/6b9a55bd-4c9c-43ef-b809-44f8fd4e9966)

**Comment by samuk** [comment]
Maybe Markdown or html with images would be good? Would it be possible to preserve the internal links?

So users get a zip of the complete wiki they could open and navigate locally? Or even upload to some generic webhosting as a static site?

**Comment by marc-farre** [comment]
I don't think there is any value exporting HTML compared to other formats.
I think links should be changed to the local file. E.g. `https://community.humhub.com/file/file/download?guid=b82974ee-df0e-4f91-bee2-2ef26afcb26d&hash_sha1=2caf7370#.jpeg` -> `file-name.jpeg`

@luke- Should the export be for a specific page or all wiki pages of a space?

**Comment by samuk** [comment]
I'd personally want this for all pages. It's the interlinked nature that gives a knowledge base value.

**Comment by luke-** [comment]
> @luke- Should the export be for a specific page or all wiki pages of a space?

I was thinking more of a single page, but with included files it gets more complicated.

**Comment by samuk** [comment]
The use case here for me is allowing users to export their wiki's in a useful way. The value of a wiki is in part it's interlinked nature so would prefer a full export if possible

**Comment by squeezer44** [comment]
The export to pdf is a good starting point in my opinion and very useful to give away single pages.

But in our environment the usecase often raise: working together on an existing wiki-sections together with many people (build social triggered content).

And from organizers point of view it would be very helpful to have the ability to trigger an export of a section or single page.
Export means: text including references and embedded objects.
Content format could be Markdown and export format could be zip.

**Comment by marc-farre** [comment]
@squeezer44 have you tried this solution?

1/ Select the section:

![image](https://github.com/user-attachments/assets/5beb2ef9-ea49-433d-9f9a-af1a77e02be3)

2/ Print:

![image](https://github.com/user-attachments/assets/754f2e53-5d21-4206-b99c-788bf8e08242)

3/ Export in PDF:

![image](https://github.com/user-attachments/assets/14f16454-956e-489f-9f0c-13269adf7dd6)

**Comment by atdqm** [comment]
In terms of functionality and user friendliness, being able to export a wiki either by page or the entire wiki would be helpful to my HumHub users. For now we're using the print to PDF but would be very happy to have a module that could do this.

**Comment by squeezer44** [comment]
@marc-farre 
I totally agree what @atdqm said: the existing PrintToPDF function is a nice gadget for regular Users to export and use single pages.
But from admins or Community Managers perspective it not covers the use case adequate - it needs more than this.
The correspondent use case is e.g.: 

As a Community Manager I need to be able to migrate WikiPageSection from one space to another space to be used in there (not possible to do this with pdf).

As an admin, I need to create a snapshot of a wiki / a wikisection for backup purpose before 20 persons of my community starts developing their wiki tomorrow morning (not possible with pdf).

**Comment by squeezer44** [comment]
Happy new year :-)
Are there any new informations regarding this topic?
Importance of this question raised.

**Comment by squeezer44** [comment]
@marc-farre 
Happy new year :-)
Do you have an update regarding this topic?
Thx for answering

**Comment by marc-farre** [comment]
Not on my side, sorry.
If we decide to create export for a single page, as suggested by @luke- , then yes, something could be done.
@samuk what do you think?

**Comment by samuk** [comment]
I don't really see the point in single page export I'm afraid. 

I've just decided not to use the wiki module.

**Comment by atdqm** [comment]
As we have already started using the wiki module and I've had requests for an export function, this subject is important to us and we would find it very useful.

**Comment by marc-farre** [comment]
@atdqm would exporting a single page be fine for you?

**Comment by atdqm** [comment]
> [@atdqm](https://github.com/atdqm) would exporting a single page be fine for you?

Yes, that would be fine for us.

**Comment by squeezer44** [comment]
Any update in here?
